File: Ejercicios/Decorator/After/Decorator.Starbuzz/Mocha.cs
namespace Decorator.Starbuzz
{
    using System;
    using System.Configuration;

	public class Mocha : CondimentDecorator
	{
		Beverage beverage;
		
		public Mocha(Beverage beverage)
		{
			this.beverage = beverage;
		}

		public override string GetDescription()
		{
			return this.beverage.GetDescription() + ", Mocha";
		}

		public override double Cost()
		{
            return .20 + this.beverage.Cost();
		}
	}
}

File: Ejercicios/2. Starbuzz (Beverages)/Java/After/src/main/starbuzz/Mocha.java
package starbuzz;

public class Mocha extends CondimentDecorator {
	Beverage beverage;
 
	public Mocha(Beverage beverage) {
		this.beverage = beverage;
	}
 
	public double cost() {
		return .20 + beverage.cost();
	}
}
